proto/identity/associations/association.proto

```diff
@@ -0,0 +1,68 @@
+syntax = "proto3";
+
+package xmtp.identity.associations;
+
+import "identity/associations/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/identity/associations";
+option java_package = "org.xmtp.proto.identity.associations";
+
+// Any address can be associated with an XID with a nonce of 0 without a
+// signature, provided that the wallet is not already associated elsewhere.
+message CreateXidUnsigned {
+  string xid = 1;
+  string initial_associated_address = 2;
+}
+
+// The first entry of any XID log. The XID must be deterministically derivable
+// from the address and nonce.
+// The recovery address defaults to the initial associated_address unless
+// there is a subsequent ChangeRecoveryAddress in the log.
+message CreateXid {
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  uint32 nonce = 3;
+  Signature associated_address_signature = 4; // TODO enforce wallets only
+}
+
+message AddAssociation {
+  enum AssociatedRole {
+    ASSOCIATED_ROLE_UNSPECIFIED = 0;
+    ASSOCIATED_ROLE_INSTALLATION = 1;
+    ASSOCIATED_ROLE_ADDRESS = 2;
+    ASSOCIATED_ROLE_LEGACY_KEY = 3; // TODO: clarify how this is used
+  }
+
+  string xid = 1;
+  uint64 client_timestamp_ns = 2;
+  AssociatedRole new_member_role = 3;
+  Signature existing_member_signature = 4;
+  Signature new_member_signature = 5;
+}
+
+// An entity may have been associated with multiple roles. Clients MUST remove
```

Comment:
When and how would an entity be associated with multiple roles?

Reply:
You could potentially associate the same keypair as a wallet address AND an installation key. I suppose we can just prohibit that

Reply:
Makes life easier if we can keep it to one. Then we can prefix all the IDs with their role when creating keys for the hashmap and categorically prevent collisions.